src/x509.cc

```diff
@@ -45,6 +45,85 @@ std::string parse_args(const Nan::FunctionCallbackInfo<v8::Value>& info) {
   return *String::Utf8Value(info[0]->ToString());
 }
 
+
+
+NAN_METHOD(verify) {
+  Nan::HandleScope scope;
+  OpenSSL_add_all_algorithms();
```

Comment:
Yeah, I was wondering about the user-friendliness of it. I'm good with
whichever since either of them will solve the problem. Just make sure that
the version is bumped to 0.3.0 when this is merged.


From: Yorkie Liu <notifications@github.com> <notifications@github.com>
Reply: Southern/node-x509
<reply@reply.github.com>
<reply@reply.github.com>
Date: September 12, 2016 at 1:19:42 PM
To: Southern/node-x509 <node-x509@noreply.github.com>
<node-x509@noreply.github.com>
Cc: Colton Baker <educability@gmail.com> <educability@gmail.com>, Comment
<comment@noreply.github.com> <comment@noreply.github.com>
Subject:  Re: [Southern/node-x509] Basic validation against certificate
authority (#38)

In src/x509.cc
<https://github.com/Southern/node-x509/pull/38#discussion_r78414738>:

> @@ -45,6 +45,85 @@ std::string parse_args(const Nan::FunctionCallbackInfo<v8::Value>& info) {
>    return *String::Utf8Value(info[0]->ToString());
>  }
>
> +
> +
> +NAN_METHOD(verify) {
> +  Nan::HandleScope scope;
> +  OpenSSL_add_all_algorithms();

Actually I'm not sure which would be better, Node.js core doesn't give the
best solution for this. Now the node-x509 and the core module crypto are in
parallel to use OpenSSL. Loading crypto at user-land would not be
user-friend in some ways?

Calling OpenSSL_add_all_algorithms() without directives looks good to me as
well now :)

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/Southern/node-x509/pull/38/files/0353b2b4e9d464638a5c7e1a2c264d559a483bdb#r78414738>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/AA2JXfXzd3sDuxQlXUx8CQvwmDuANunDks5qpYmugaJpZM4Jmuix>
.